feat(rust): add ockam_api_nodes

twittner opened the pull request:
On top of #2767.

twittner committed:
feat(rust): add ockam_api_nodes

mrinalwadhwa commented:
I'm not sure we should have a separate crate for `ockam_api_nodes`, could that be a module instead?

twittner commented:
It is possible to fold `ockam_api_nodes` into `ockam_api`. I thought that we might want to share the core type definitions with other crates, so I wanted to keep the number of dependencies minimal, but we can instead also have one api crate that contains all implementations if that is preferred.

twittner committed:
feat(rust): add command-line interface for nodes api

twittner committed:
feat(rust): add builders to ockam_api

To enable a more fluent construction of requests and responses and
their optional bodies.

twittner commented:
Includes https://github.com/build-trust/ockam/pull/2778.
